dspace-api/src/main/java/org/dspace/app/util/SyndicationFeed.java

```diff
@@ -257,8 +257,11 @@ public void populate(HttpServletRequest request, Context context, IndexableObjec
                 String pubDateString = getOneDC(item, dateField);
                 if (pubDateString != null) {
                     ZonedDateTime pubDate = new DCDate(pubDateString).toDate();
-                    entry.setPublishedDate(java.util.Date.from(pubDate.toInstant()));
-                    hasDate = true;
+                    // If date string could not be parsed as a date, then pubDate will be null
+                    if (pubDate != null) {
+                        entry.setPublishedDate(java.util.Date.from(pubDate.toInstant()));
+                        hasDate = true;
+                    }
```

Comment:
That will solve the immediate problem, but should we log a WARN here to help the admin. who needs to discover why published-dates are disappearing?

Reply:
i agree, especially since this is a case where there is a value in the pubDateString, just apparently not a valid one

Reply:
@mwoodiupui and @kshepherd : A warning is already logged by `DCDate` just above this line of code.  Just before this `if` is this line:
```
ZonedDateTime pubDate = new DCDate(pubDateString).toDate();
```

If that line results in `pubDate` being `null`, then `DCDate` will already log a warning here: 
https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/content/DCDate.java#L243-L244

So, by adding another warning, we'll now see two logged warnings in a row.

Reply:
I guess the only benefit of the *extra* logged warning (which @kshepherd added) is that it will reference *which* Item is having the date parsing error.  But, we definitely will have two logged warnings for each invalid date.

Reply:
I've decided to leave the duplicate logs, as the second log can provide more information on which Item & which field is invalid. I pushed up a minor checkstyle fix & added the date field name to the log (as the date field is configurable).

Reply:
I didn't catch that, just revert/drop my commit if it's going to add log spam and you'd rather leave it